shared/middleware/animation_library_api.rb

```diff
@@ -162,14 +162,14 @@ class AnimationLibraryApi < Sinatra::Base
   end
 
   #
-  # POST /api/v1/animation-library/default-spritelab-metadata
+  # POST /api/v1/animation-library/default-spritelab-metadata/(levelbuilder|production)
   #
-  # Update default sprite JSON in S3
-  post %r{/api/v1/animation-library/default-spritelab-metadata} do
+  # Update default sprite JSON in S3 for the given environment
+  post %r{/api/v1/animation-library/default-spritelab-metadata/(levelbuilder|production)} do |env|
     dont_cache
     if request.content_type == 'application/json'
       body = request.body.string
-      key = ANIMATION_DEFAULT_MANIFEST_JSON_LEVELBUILDER
+      key = (env == "production") ? ANIMATION_DEFAULT_MANIFEST_JSON : ANIMATION_DEFAULT_MANIFEST_JSON_LEVELBUILDER
```

Comment:
we could future-proof this logic with a small update to this logic. defaulting to levelbuilder is definitely the safer behavior, but might not be expected if we were to add an environment option to this endpoint. we could manage these paths in a hash to prevent unexpected behavior:

```ruby
ANIMATION_MANIFEST = {
  levelbuilder: ANIMATION_DEFAULT_MANIFEST_JSON_LEVELBUILDER,
  production: ANIMATION_DEFAULT_MANIFEST_JSON
}.freeze

key = ANIMATION_MANIFEST[env]
if (!key) {
  # log an error to honeybadger, return a 404, etc.
  return
}
```

Reply:
Ruby Question - the path for this api is `post %r{/api/v1/animation-library/default-spritelab-metadata/(levelbuilder|production)} do |env|`. Does this allow an `env` that is not `levelbuilder` or `production`?

Reply:
no, the request will 404 if you pass a different environment. i'm thinking about the case where we add to this list of environments -- it would be unexpected that `POST /api/v1/animation-library/default-spritelab-metadata/test` would upload to levelbuilder by default rather than no-oping or returning an error